Next.mk

```diff
@@ -37,32 +39,97 @@ build: $(current_binary) ## Build the func-e binary
 test: ## Run all unit tests
 	@printf "$(ansi_format_dark)" test "running unit tests"
 	@go test . ./internal/...
-	@printf "$(ansi_format_bright)" test "✔"
+	@printf "$(ansi_format_bright)" test "ok"
 
 # Tests run one at a time, in verbose mode, so that failures are easy to diagnose.
 # Note: -failfast helps as it stops at the first error. However, it is not a cacheable flag, so runs won't cache.
 e2e: $(current_binary) ## Run all end-to-end tests
 	@printf "$(ansi_format_dark)" e2e "running end-to-end tests"
 	@E2E_FUNC_E_PATH=$(dir $(current_binary)) go test -parallel 1 -v -failfast ./e2e
-	@printf "$(ansi_format_bright)" e2e "✔"
+	@printf "$(ansi_format_bright)" e2e "ok"
 
-# Use patterns to set appropriate variables needed for multi-platform builds
-$(build_dir)/func-e_darwin_%/func-e: override goos=darwin
-$(build_dir)/func-e_linux_%/func-e: override goos=linux
-$(build_dir)/func-e_%_amd64/func-e: override goarch=amd64
-$(build_dir)/func-e_%_arm64/func-e: override goarch=arm64
-$(build_dir)/func-e_%/func-e:
-	$(call go-build, $@)
+# TODO: arm64 on Windows https://github.com/envoyproxy/envoy/issues/17572
+platforms := darwin_amd64 darwin_arm64 linux_amd64 linux_arm64 windows_amd64
 
-$(windows_binary): override goos=windows
-$(windows_binary): override goarch=amd64
-$(windows_binary):
-	$(call go-build, $@)
+$(build_dir)/func-e_%/func-e: main.go
+	$(call go-build, $@, $^)
 
+$(build_dir)/func-e_%/func-e.exe: main.go
```

Comment:
in case you didn't know the nuance of make, you can only use a '%' once, that's why we have to do this twice